core/src/main/scala/org/apache/spark/internal/config/Deploy.scala

```diff
@@ -39,6 +41,14 @@ private[spark] object Deploy {
     .checkValues(RecoverySerializer.values.map(_.toString))
     .createWithDefault(RecoverySerializer.JAVA.toString)
 
+  val RECOVERY_COMPRESSION_CODEC = ConfigBuilder("spark.deploy.recoveryCompressionCodec")
+    .doc("A compression codec for persistence engines. lz4 (default), lzf, snappy, and zstd. " +
```

Comment:
How the end users specify no compression codec? Seems like we would need sth like `none` and/or `uncompressed`.